source/partials/header/_link.html.erb

```diff
@@ -0,0 +1,30 @@
+<%
```

Comment:
I'm not a huge fan of these blocks of ruby logic at the top of erb files. Is there some way we can move some to helpers?

Reply:
Umm, can you?  There's no _automagical helper_ for a partial—this isn't a Rails view?

I could see moving this to a helper, like (I'm not even sure instance variables work here):
```ruby
<%
  HeaderHelper::build_link_ivars(locals)
  # now you have some global @vars?
%>

<a href="<%= @link[:href] %>"  …>
```

Or with no instance variables, but then does it make sense?
```ruby
<%
  link = locals[:link]
  show_description = (mobile: locals[:mobile], link: link)
%>
```

I _think_ we're on the same page that we'd prefer `<% %>` at the top over inline `.html.erb` logic?  But yeah, I don't know where to put it otherwise 😟 with raw Ruby.  I could be overlooking something too!

Reply:
I believe there is the ability to load custom helpers and use them in views: https://middlemanapp.com/basics/helper-methods/

Could then use some of those to perhaps package some repeated logic across views. But I don't think its an important factor in this PR, this is following the pattern of previous pages so its definitely non-blocking.

Main thing, as you pointed out, is ease of testing.

Reply:
None of this is repeated at all and you have to pass everything down via `locals`.  There is no stack context to inject into here.  Even with instance variables, I found a few edge cases where they were _bleeding_ across different pages when building all at once, so they're a bit worthless..

I couldn't see any value in migrating the logic into a helper upon further review.  I think the unit tests for them just provide false coverage as you still have to map `locals` down 10 views to get into the helper.  You could mutate just about anything—forget to pass `mobile: true` or `locals[:link]`—and unit tests would still pass and CI wouldn't fail, so the coverage seems useless.

Instead, I just added into the Feature test as I felt it was required either way: 6b0ff3e5